dali/python/nvidia/dali/plugin/jax/iterator.py

```diff
@@ -252,54 +252,101 @@ def data_iterator_impl(
         last_batch_padded=False,
         last_batch_policy=LastBatchPolicy.FILL,
         prepare_first_batch=True,
-        sharding=None):
+        sharding=None,
+        devices=None):
```

Comment:
This is the place where actual functionality is added. This function is a common implementation used in different decorators later.